api/src/github-user/types.ts

```diff
@@ -0,0 +1,9 @@
+import {} from "module";
+import { GeneralResponseDto } from "../app/types";
+import { GitHubUserApiResponse } from "../github/types";
+import { ValidateNested } from "class-validator";
+
+export class GetUserResponseDto extends GeneralResponseDto {
+  @ValidateNested({ each: true })
```

Comment:
Since user is only one object, and not an array, `{ each: true }` option is not needed, otherwise it will showup as an array in swagger docs :p

```suggestion
  @ValidateNested()
```